### ISSUE TITLE
Question Regarding LLAVA-LoRA baseline

### ISSUE THREAD

**Issue by ghost** [Hoar012/RAP-MLLM#1]
I have a few questions regarding the llava-lora baseline:

1. For the training, do we use a single lora over all the user's concepts ? or just single lora for each concept and then evaluate on test examples.

2. During training do we use a trigger word ? e.g. <sks> dog, <sbu> man etc. Can you provide some examples of the captioning used ?

**Comment by Hoar012** [comment]
Thanks for your interest in our work. 

1. Our desired personalized model should be able to remember and recognize all concepts simultaneously. This way, even if the user provides an image without specifying the target concept, the model can still recognize relevant concepts and generate appropriate responses. To achieve this goal, LLaVA-LoRA needs to learn all concepts within a single LoRA.

2. If we want the model to learn the concept <my_cat>, we should prepare training images containing <my_cat>, and the corresponding captions should also include the phrase <my_cat>.  
For instance, a training sample could consist of: <br><img src="https://github.com/user-attachments/assets/2c92e44d-1fa2-41f5-b452-cbe621c5a659" alt="my_cat" width="260"><br> with the query "Give a caption of this image." The desired output would be: "<my_cat> sleepily laying on a gray drying mat in front of a kitchen window".  
More details of the dataset can be found in MyVLM's collection at [MyVLM Dataset on Hugging Face](https://huggingface.co/datasets/yuvalalaluf/MyVLM).

**Comment by roberto545** [comment]
Thanks for prompt response. Regarding the hyper-parmeters for LLAVA-LORA, could you share again like rank, attachment points etc ? It is not explicitly mentioned in the paper.

**Comment by Hoar012** [comment]
I have uploaded the script used for training. You can check the hyperparameters used in the experiment in `/scripts/train_lora_llava.sh`.

**Comment by roberto545** [comment]
Thanks. I ran this. Seems to a bug at. So 'prepare dataset' is not uploaded fully ?

list_data_dict = prepare_dataset(list_data_dict)  # TODO prepare dataset

**Comment by Hoar012** [comment]
I have checked this function, and it runs fine. If you want to train a LLaVA-LoRA model, you need to provide a dataset in the LLaVA format: 
```
[{
        "id": 0,
        "image": path_to_image,
        "conversations": [
            {
                "from": "human",
                "value": "<image>\nGive a caption of this image."
            },
            {
                "from": "gpt",
                "value": expected_caption
            }
        ]
}]```

**Comment by roberto545** [comment]
Thanks for the response. When training the RAP-MLMM on the personalized dataset, the only parameters that are trained are lora parameters right ?

**Comment by Hoar012** [comment]
Yes, after training the model, you will obtain a LoRA weight. You can also merge it with the original model.

**Comment by Hoar012** [comment]
Yes.
